src/OmnichannelChatSDK.ts

```diff
@@ -2673,6 +2684,66 @@ class OmnichannelChatSDK {
             }
         }
     }
+
+    private async cleanupFailedConversation(error: Error, liveChatContext?: StartChatOptionalParams['liveChatContext']): Promise<void> {
+
+        this.scenarioMarker.startScenario(TelemetryEvent.CleanupFailedStartchat);
+
+        const isLivechatContextPresent = Boolean(liveChatContext && Object.keys(liveChatContext).length > 0);
+        const isReconnectIdPresent = Boolean(this.reconnectId && this.reconnectId.trim().length > 0);
+
+        /**
+         * Only cleanup if it's a MessagingClientConversationJoinFailure on a freshly created conversation
+         *
+         * DO NOT continue if:
+         * - The error is not a ChatSDKError or not related to conversation join failure
+         * - The conversation is not freshly created (i.e., if `useCreateConversation` is disabled)
+         * - The conversation was previously created (i.e., if `isLivechatContextPresent` is true)
+         * - The error is related to a reconnect attempt (i.e., if `isReconnectIdPresent` is true)
+         */
+
+        if (!(error instanceof ChatSDKError &&
+            error?.message === ChatSDKErrorName.MessagingClientConversationJoinFailure &&
+            !this.chatSDKConfig.useCreateConversation?.disable &&
+            !isLivechatContextPresent &&
+            !isReconnectIdPresent)) {
+            return;
+        }
+
+        try {
+            const sessionCloseOptionalParams: ISessionCloseOptionalParams = {};
+            if (this.authenticatedUserToken) {
+                sessionCloseOptionalParams.authenticatedUserToken = this.authenticatedUserToken;
+            }
+
+            await this.OCClient.sessionClose(this.requestId, sessionCloseOptionalParams);
```

Comment:
When a `sessionClose()`, isn't there any additional internal attributes that need to be cleaned up as well?